pkg/helm/chart_info.go

```diff
@@ -146,23 +146,19 @@ func parseChartRequirementsFile(chartDirectory string, apiVersion string) (Chart
 	return chartRequirements, nil
 }
 
-func removeIgnored(rootNode *yaml.Node, rootKind yaml.Kind) {
-	var toDelete []int
-	for i, node := range rootNode.Content {
-		if strings.Contains(node.HeadComment, "# -- @ignore") {
-			toDelete = append(toDelete, i)
-			if rootKind == yaml.MappingNode {
-				toDelete = append(toDelete, i+1)
-			}
+func removeIgnored(rootNode *yaml.Node, parentKind yaml.Kind) {
+	newContent := make([]*yaml.Node , 0, len(rootNode.Content))
+	for i := 0; i < len(rootNode.Content); i++ {
+		node := rootNode.Content[i]
+		if !strings.Contains(node.HeadComment, "@ignore") {
+			removeIgnored(node, node.Kind)
+			newContent = append(newContent, node)
+		} else if parentKind == yaml.MappingNode {
```

Comment:
This can be reworded to:
if HeadComment contains `@ignore` don't include it in newContent. If root node is `MappingNode` don't include the next  node as well

Reply:
oh! I see what you're saying...